tests/monad.cpp

```diff
@@ -17,7 +22,33 @@ static auto unsafe_divide_4_by(double val) {
     return 4.0 / val;
 }
 
-TEST_CASE("rsl::mbind") {
+template <typename T>
+using result_t = tl::expected<T, std::string>;
+
+static result_t<double> divide(double x, double y) {
+    if (y == 0) {
+        return tl::make_unexpected(std::string{"divide by 0"});
+    }
+    return x / y;
+}
+
+static result_t<double> multiply(double x, double y) { return x * y; }
+
+static result_t<double> divide_3(double x) { return divide(3, x); }
+
+static result_t<double> multiply_3(double x) { return multiply(3, x); }
+
+template <typename OUT, typename IN>
```

Comment:
```suggestion
template <typename Out, typename In>
```
Or maybe just `U` and `T`.

Reply:
removed this function, wasn't being called anywhere